test/InputOutput/unit_read_value_type.jl

```diff
@@ -0,0 +1,54 @@
+using Test
+import ClimaCore
+
```

Comment:
```suggestion
compare_read_type(x) = InputOutput.read_type(string(eltype(x))) == eltype(x)
```
Let's define a helper above and then use `@test compare_read_type(ClimaCore.Fields.local_geometry_field(space))`